### PR TITLE
Add Campaign Selector AT-516

### PR DESCRIPTION
**Pull request by federicoweber**
### Purpose
To add a dropdown to fetch and select campaigns.

### Notes
This also add the logic to fetch the campaigns (_for testing purpose we are returning mocked data_).
[ref](https://buffer.atlassian.net/jira/software/projects/AT/boards/69?selectedIssue=AT-516)

### TODO
- [ ] update url structure to `\campaigns\<CAMPAIGN_ID>`

#### Staging Deployment

_This repo is CI/CD enabled and staging deployment should be available at:
https://<branch_name>.analyze.buffer.com :smile:_

### CODE REVIEW

File: packages/campaign-selector/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# @bufferapp/add-report
```
Comment (dinostheo) [inline packages/campaign-selector/README.md]: Non-blocking and very minor, but the README needs to be adjusted.

File: packages/campaign-selector/reducer.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    default:
+      return state;
+  }
+};
+
+export const actions = {
+  select(campaign) {
+    return {
+      type: actionTypes.SELECT,
+      id: campaign.id,
```
Comment (dinostheo) [inline packages/campaign-selector/reducer.js]: I will adjust the campaign pulse chart to this structure 😃

File: packages/campaign-selector/reducer.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  campaigns: [],
+  selectedCampaign: null,
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case actionTypes.SELECT:
+      return {
+        ...state,
+        selectedCampaign: getSelecetedCampaign(state, action.id),
```
Comment (gmzjuliana) [inline packages/campaign-selector/reducer.js]: small typo here and on line 8 :) getSelectedCampaign

Comment (federicoweber) [reply packages/campaign-selector/reducer.js]: great catch thanks @gmzjuliana 🙌

**Pull request merged**